CRM/Financial/BAO/Payment.php

```diff
@@ -103,61 +103,81 @@ public static function create(array $params): CRM_Financial_DAO_FinancialTrxn {
       // Record new "payment" (financial_trxn, financial_item, entity_financial_trxn etc).
       $salesTaxFinancialAccount = CRM_Contribute_BAO_Contribution::getSalesTaxFinancialAccounts();
       // Get all the lineitems and add financial_item information to them for the contribution on which we are recording a payment.
-      $financialItems = LineItem::get(FALSE)
-        ->addSelect('tax_amount', 'price_field_value_id', 'financial_item.id', 'financial_item.status_id:name', 'financial_item.financial_account_id', 'financial_item.entity_id')
+      // @todo - get this payableItems array in the first place from getPayableItems() above.
+      $items = LineItem::get(FALSE)
+        ->addSelect('*', 'financial_item.status_id:name', 'financial_item.id', 'financial_item.financial_account_id', 'financial_item_id.currency', 'financial_item.financial_account_id.is_tax', 'financial_item.entity_id')
         ->addJoin(
           'FinancialItem AS financial_item',
-          'INNER',
+          'LEFT',
           NULL,
```

Comment:
FYI This line can be deleted, the `addJoin()` function ignores NULL values.

Reply:
@colemanw but it's not the last argument - & I need those last 2 I think?

```
      ->addJoin(
        'FinancialItem AS financial_item',
        'LEFT',
        NULL,
        ['financial_item.entity_table', '=', '"civicrm_line_item"'],
        ['financial_item.entity_id', '=', 'id']
      )
```

Reply:
@colemanw no I'm confused - this is the function signature - are you saying I can pass `$conditions` without passing the preceding arguments & is so are you saying that would be LESS confusing?

![image](https://github.com/civicrm/civicrm-core/assets/336308/2489bb44-ef18-457d-a9d4-1f8dd26364ec)

Reply:
Yea it's a little bit black-magicy but you can see the function basically just stuffs all the params into an array (using `array_unshift()`) so the result is the same whether you pass NULL for the 3rd param or just skip it and the 3rd param is your first condition. If you follow the logic of the function, this input:
`->addJoin('Foo', 'INNER', NULL, ['key1', '=', 'val1'], ['key2', '=', 'val2'])`
will yield the following result:
`['Foo', 'INNER', ['key1', '=', 'val1'], ['key2', '=', 'val2']]`
but *this* input:
`->addJoin('Foo', 'INNER', ['key1', '=', 'val1'], ['key2', '=', 'val2'])`
will yield the exact same result:
`['Foo', 'INNER', ['key1', '=', 'val1'], ['key2', '=', 'val2']]`
Because in the first case `if ($bridge)` was false so it skipped null. In the second case it was true so it added it to the array. Either way works out the same.

Reply:
@eileenmcnaughton maybe this would help: https://github.com/civicrm/civicrm-core/pull/29899

Reply:
@colemanw I dunno - still super confusing that you are basically passing something that is NOT `$type` for `$type`

Maybe if you removes those & made it `addJoin($entity, $conditions....)` then the black magic would be grey